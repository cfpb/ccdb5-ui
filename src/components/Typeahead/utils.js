/**
 * Sets the options for the async typeahead
 *
 * @param {string} value - value typed into the typeahead
 * @param {Function} setOptions - function to set local array of options
 * @param {string} uri - the endpoint being hit by the fetch
 * @returns {void | Promise<object>} - Either nothing or a response with option object
 */
export function handleFetchSearch(value, setOptions, uri) {
  const num = value.toLowerCase();
  if (num === '') {
    setOptions([]);
    return;
  }
  return fetch(uri)
    .then((result) => result.json())
    .then((items) => {
      const options = items.map((item) => ({
        key: item,
        label: item,
        position: item.toLowerCase().indexOf(num),
        value,
      }));
      setOptions(options);
    });
}
