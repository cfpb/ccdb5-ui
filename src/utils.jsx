import { DATE_RANGE_MIN, SLUG_SEPARATOR } from './constants'
import moment from 'moment'

// eslint-disable-next-line complexity
export const calculateDateInterval = ( minDate, maxDate ) => {
  // only check intervals if the end date is today
  // round off the date so the partial times don't mess up calculations
  const today = moment( new Date() ).startOf( 'day' )
  const end = moment( maxDate ).startOf( 'day' )
  const start = moment( minDate ).startOf( 'day' )

  // make sure end date is the same as today's date
  if ( end.diff( today, 'days', true ) !== 0 ) {
    return ''
  }

  // is the start date the same as the oldest document?
  if ( moment( minDate ).isSame( DATE_RANGE_MIN, 'day' ) ) {
    return 'All'
  }

  // verify if it's 3 or 1 years
  const yrDiff = end.diff( start, 'years', true )

  if ( yrDiff === 3 || yrDiff === 1 ) {
    return yrDiff + 'y'
  }

  // verify if it's 6 or 3 months
  const moDiff = end.diff( start, 'months', true )
  if ( moDiff === 6 || moDiff === 3 ) {
    return moDiff + 'm'
  }

  return ''
}

/**
 * Function to set the limit of the range of a set of numbers
 * @param {int} x value we are checking
 * @param {int} min smallest number it can me
 * @param {int} max biggest number it can be
 * @returns {*}the limited value
 */
export const clamp = ( x, min, max ) => {
  if ( x < min ) {
    x = min;
  } else if ( x > max ) {
    x = max;
  }
  return x;
}

/**
 * Replacement for the common pattern:
 * if( o.field )
 *    x = o.field
 * else
 *    x = alternateValue
 *
 * Avoids some of the complexity lint warnings
 *
 * @param {Object} o the object being tested
 * @param {string} field the field to check
 * @param {string|Object} alternateValue the value to use in absence
 * @returns {string} the value to use
 */
export const coalesce = ( o, field, alternateValue ) => {
  if ( typeof o !== 'object' ) {
    return alternateValue;
  }

  return field in o && o[field] ? o[field] : alternateValue;
};

/**
 * Creates a hash from a string
 *
 * @param {string} someString the string to hash
 * @returns {int} a hashing of the string
 */
export function hashCode( someString ) {
  const s = String( someString )
  let hash = 0
  let i, chr
  if ( s.length === 0 ) {
    return hash
  }
  for ( i = 0; i < s.length; i++ ) {
    chr = s.charCodeAt( i )
    hash = ( hash << 5 ) - hash + chr

    // Convert to 32bit integer
    hash |= 0
  }
  return hash
}

/**
 * Creates a hash from an object
 *
 * @param {string} o the object to hash
 * @returns {string} a hashing of the object
 */
export function hashObject( o ) {
  return hashCode( JSON.stringify( o ) )
}
export const normalize = s => s.toLowerCase()

export const slugify = ( a, b ) => a + SLUG_SEPARATOR + b

export const sortSelThenCount = ( options, selected ) => {
  const retVal = ( options || [] ).slice()

  // Sort the array so that selected items appear first, then by doc_count
  /* eslint complexity: ["error", 5] */
  retVal.sort( ( a, b ) => {
    const aSel = selected.indexOf( a.key ) !== -1
    const bSel = selected.indexOf( b.key ) !== -1

    if ( aSel && !bSel ) {
      return -1;
    }
    if ( !aSel && bSel ) {
      return 1;
    }

    // Both are selected or not selected
    // Sort by descending doc_count
    return b.doc_count - a.doc_count
  } )

  return retVal
}

/**
* Safely format a date
*
* @param {Date} date the date to convert
* @returns {string} the date formatted for the current locale
*/
export function shortFormat( date ) {
  const wrapped = moment( date )
  return date ? wrapped.format( 'M/D/YYYY' ) : ''
}

/**
* Convert a date to a truncated ISO-8601 string
*
* @param {Date} date the date to convert
* @returns {string} the date formatted as yyyy-mm-ddd
*/
export function shortIsoFormat( date ) {
  return date ? date.toISOString().substring( 0, 10 ) : ''
}

// ----------------------------------------------------------------------------
// attribution: underscore.js (MIT License)

/**
* Delay the implementation of a function until after a period of time
* This prevents expensive calls from being made while triggering events are
* still happening
*
* @param {function} func a function with an embedded expensive call
* @param {int} wait the number of msecs to delay before calling the function
* @returns {function} a replacement function to use in place of the original
*/
export function debounce( func, wait ) {
  var timer = null;

  var later = ( context, args ) => {
    timer = null;
    func.apply( context, args );
  }

  return ( ...theArgs ) => {
    if ( !timer ) {
      timer = setTimeout( later, wait, this, theArgs )
    }
  }
}

// ----------------------------------------------------------------------------
// attribution: lodash.js (Creative Commons License)

/**
* Binds methods of an object to the object itself, overwriting the existing
* method
*
* @param {Object} obj The object to bind and assign the bound methods to.
* @param {...(string|string[])} methodNames The object method names to bind,
*  specified individually or in arrays.
* @returns {Object} the updated object
*/
export function bindAll( obj, methodNames ) {
  const length = methodNames.length
  for ( let i = 0; i < length; i++ ) {
    const key = methodNames[i]
    obj[key] = obj[key].bind( obj )
  }
  return obj;
}
