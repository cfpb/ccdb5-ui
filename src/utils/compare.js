export const arrayEquals = (first, second) => {
  let length = first.length;
  if (length !== second.length) return false;
  while (length--) {
    if (first[length] !== second[length]) return false;
  }
  return true;
};

/**
 * Function to compare two objects or arrays.
 *
 * @param {object | Array | any} first - value 1 to convert to string
 * @param {object | Array | any} second - value 2 to convert to string to compare
 * @returns {boolean} Whether the values are equal
 */
export const isEqual = (first, second) =>
  JSON.stringify(first) === JSON.stringify(second);
