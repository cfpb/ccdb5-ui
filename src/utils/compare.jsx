export const arrayEquals = (a, b) => {
  let i = a.length;
  if (i !== b.length) return false;
  while (i--) {
    if (a[i] !== b[i]) return false;
  }
  return true;
};
