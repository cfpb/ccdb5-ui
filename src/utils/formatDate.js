// ----------------------------------------------------------------------------
// Exports
import dayjs from 'dayjs';
import dayjsLocalizedFormat from 'dayjs/plugin/localizedFormat';
import dayjsUtc from 'dayjs/plugin/utc';

dayjs.extend(dayjsLocalizedFormat);
dayjs.extend(dayjsUtc);

/**
 * Function to format/convert a string to format we want
 *
 * @param {(string | object)} uglyDate - the input string to convert
 * @returns {string} the cleaned up string in YYYY-MM-DD
 */
export const formatDate = (uglyDate) => {
  if (!uglyDate || (typeof uglyDate === 'string' && uglyDate.length === 10)) {
    return uglyDate;
  }
  return dayjs(new Date(uglyDate)).format('YYYY-MM-DD');
};

export const formatDisplayDate = (dateString) => {
  return dayjs(new Date(dateString)).utc().format('M/D/YYYY');
};
/**
 * adjusting dates coming from the charts so the dates are correct
 *
 * @param {(string | object)} dateIn - the input string to convert
 * @returns {string} the cleaned up string in M/D/YYYY
 */
export const adjustDate = (dateIn) =>
  dayjs(new Date(dateIn)).utc().add(5.5, 'hours').format();

/**
 * Function to format/convert a string to format we want for the model
 *
 * @param {(string | object)} dateIn - the input string to convert
 * @returns {string} the cleaned up string in YYYY-MM-DD
 */
export const formatDateModel = (dateIn) =>
  dayjs(new Date(dateIn)).utc().add(5.5, 'hours').format('YYYY-MM-DD');

export const formatDateIso = (dateIn) => dayjs(dateIn).toISOString();
/**
 * Function to format/convert a string to format we want for the model
 *
 * @param {(string | object)} dateIn - the input string to convert
 * @returns {string} the cleaned up string in Jul 4, 2010
 */
export const formatDateLocaleShort = (dateIn) =>
  dayjs(new Date(dateIn)).utc().add(5.5, 'hours').format('ll');

/**
 * function to convert and compare 2 strings as dates
 *
 * @param {string} date1 - input date string to compare MM/DD/YYYY or YYYY-MM-DD
 * @param {string} date2 - input date string to compare MM/DD/YYYY or YYYY-MM-DD
 * @returns {boolean} lets us know if date is equal
 */
export const isDateEqual = (date1, date2) =>
  dayjs(new Date(date1)).isSame(new Date(date2), 'day');

/**
 * function to convert and compare 2 strings as dates
 *
 * @param {string} date1 - input date string to compare MM/DD/YYYY or YYYY-MM-DD
 * @param {string} date2 - input date string to compare MM/DD/YYYY or YYYY-MM-DD
 * @returns {number} the return of the compared converted values
 */
export const compareDates = (date1, date2) => {
  if (isDateEqual(date1, date2)) {
    return 0;
  }

  return new Date(date1) < new Date(date2) ? -1 : 1;
};
