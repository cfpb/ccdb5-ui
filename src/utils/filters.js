import { filterPatch, SLUG_SEPARATOR } from '../constants';
import { slugify } from '../utils';

export const formatPillPrefix = (fieldName) => {
  // update this if they want the pill prefixes in other fields.
  if (fieldName === 'timely') {
    const rep = /_/g;
    const prefix = fieldName.replace(rep, ' ');
    return prefix[0].toUpperCase() + prefix.slice(1) + ': ';
  }
  return '';
};

export const getUpdatedFilters = (filterName, filters, aggs, fieldName) => {
  // early exit if its not issue or product
  if (!filterPatch.includes(fieldName)) {
    return filters;
  }

  // remove parent filter
  const parts = filterName.split(SLUG_SEPARATOR);
  const parentFilter = parts[0];
  const hasParent = filters.includes(parentFilter);
  // remove current filter
  const oldFilters = filters
    .filter((filter) => filter !== parentFilter && filterName)
    .filter((filter) => filter !== filterName);
  // apply siblings
  const sibs = [];
  if (hasParent) {
    const siblings = aggs.find((agg) => agg.key === parentFilter);
    siblings['sub_' + fieldName + '.raw'].buckets.forEach((bucket) => {
      // don't include self
      if (bucket.key !== parts[1]) {
        sibs.push(slugify(parentFilter, bucket.key));
      }
    });
  }

  return oldFilters.concat(sibs);
};
