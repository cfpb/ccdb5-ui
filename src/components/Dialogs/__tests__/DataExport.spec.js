jest.mock('../../../actions/dataExport');

import { shallow } from 'enzyme';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ReduxDataExport, {
  DataExport,
  mapDispatchToProps,
} from '../DataExport';
import * as utils from '../../../utils';

const mockDataExportActions = require('../../../actions/dataExport');

function setupEnzyme() {
  const props = {
    allComplaints: 999,
    exportAll: jest.fn(),
    exportSome: jest.fn(),
    someComplaints: 99,
    queryState: { foo: 'bar' },
  };

  const target = shallow(<DataExport {...props} />);

  return {
    props,
    target,
  };
}

function setupSnapshot(total = 1001) {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    aggs: {
      doc_count: 9999,
      total,
    },
    query: {
      tab: 'foo',
    },
  });

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxDataExport />
      </IntlProvider>
    </Provider>
  );
}

describe('component::DataExport', () => {
  beforeAll(() => {
    document.queryCommandSupported = jest.fn((_) => true);
  });

  describe('initial state', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot();
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('hides the dataset radio buttons when there is no filter', () => {
      const target = setupSnapshot(9999);
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('visual changes', () => {
    it('switches the button after a copy', () => {
      const target = setupSnapshot();
      const ctlDataExport = target.root.findByType(DataExport);
      ctlDataExport.instance.setState({
        copied: true,
        dataset: 'full',
        exportUri: 'http://www.example.org',
        format: 'csv',
      });

      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('user interaction', () => {
    let target, props;
    beforeEach(() => {
      ({ target, props } = setupEnzyme());
    });

    it('supports changing the format', () => {
      const radio = target.find('#format_csv');
      radio.simulate('change', { target: radio.getElements()[0].props });
      expect(target.state('format')).toEqual('csv');
      expect(target.state('copied')).toEqual(false);
    });

    it('supports changing the which dataset is used', () => {
      const radio = target.find('#dataset_filtered');
      radio.simulate('change', { target: radio.getElements()[0].props });
      expect(target.state('dataset')).toEqual('filtered');
      expect(target.state('copied')).toEqual(false);
    });

    describe('clicking Copy to Clipboard', () => {
      let copyToClipboardBtn;
      beforeEach(() => {
        copyToClipboardBtn = target.find('.heres-the-url button').first();
      });

      it('is disabled at the beginning', () => {
        expect(copyToClipboardBtn.prop('disabled')).toEqual(true);
      });

      it('calls a specific action when "filtered" is chosen', () => {
        const el = { select: jest.fn() };
        const ev = { target: { focus: jest.fn() } };
        document.execCommand = jest.fn();
        document.getElementById = jest.fn((_) => el);

        target.setState({ dataset: 'filtered', format: 'csv' });
        copyToClipboardBtn.simulate('click', ev);

        expect(document.getElementById).toHaveBeenCalledWith('exportUri');
        expect(document.execCommand).toHaveBeenCalledWith('copy');
        expect(el.select).toHaveBeenCalledWith();
        expect(ev.target.focus).toHaveBeenCalledWith();
        expect(target.state('copied')).toEqual(true);
      });
    });

    describe('clicking Start Export', () => {
      let startExport;
      beforeEach(() => {
        startExport = target.find('.footer button').first();
      });

      it('is disabled at the beginning', () => {
        expect(startExport.prop('disabled')).toEqual(true);
      });

      it('calls a specific action when "filtered" is chosen', () => {
        target.setState({ dataset: 'filtered' });
        startExport.simulate('click');
        expect(props.exportSome).toHaveBeenCalled();
      });

      it('calls a specific action when "all" is chosen', () => {
        target.setState({ dataset: 'full' });
        startExport.simulate('click');
        expect(props.exportAll).toHaveBeenCalled();
      });

      it('displays the long download warning', () => {
        target.setState({ dataset: 'full' });
        startExport.simulate('click');
        expect(target.state('mode')).toEqual('NOTIFYING');
      });
    });
  });

  describe('getDerivedStateFromProps', () => {
    let target, state, props;
    beforeEach(() => {
      ({ target, props } = setupEnzyme());

      state = {
        format: 'csv',
      };
    });

    describe('when dataset == full', () => {
      beforeEach(() => {
        state.dataset = 'full';
      });

      it('builds a specific URL', () => {
        DataExport.getDerivedStateFromProps(props, state);
        expect(mockDataExportActions.buildAllResultsUri).toHaveBeenCalledWith(
          'csv'
        );
      });
    });
    describe('when dataset == filtered', () => {
      beforeEach(() => {
        state.dataset = 'filtered';
      });

      it('builds a different URL', () => {
        DataExport.getDerivedStateFromProps(props, state);
        expect(mockDataExportActions.buildSomeResultsUri).toHaveBeenCalledWith(
          'csv',
          99,
          { foo: 'bar' }
        );
      });
    });
  });

  describe('mapDispatchToProps', () => {
    let dispatch, gaSpy;
    beforeEach(() => {
      dispatch = jest.fn();
      gaSpy = jest.spyOn(utils, 'sendAnalyticsEvent');
    });

    afterEach(() => {
      jest.clearAllMocks();
    });
    it('provides a way to call exportAllResults', () => {
      mapDispatchToProps(dispatch).exportAll('json', 'mytab');
      expect(dispatch.mock.calls.length).toEqual(1);
      expect(gaSpy).toHaveBeenCalledWith('Export All Data', 'mytab:json');
    });

    it('provides a way to call exportSomeResults', () => {
      mapDispatchToProps(dispatch).exportSome('csv', 1300, 'atab');
      expect(dispatch.mock.calls.length).toEqual(1);
      expect(gaSpy).toHaveBeenCalledWith('Export Some Data', 'atab:csv');
    });
  });
});
