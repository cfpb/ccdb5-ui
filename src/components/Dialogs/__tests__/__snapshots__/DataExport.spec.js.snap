// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::DataExport initial state hides the dataset radio buttons when there is no filter 1`] = `
<section
  className="export-modal"
>
  <div
    className="header layout-row"
  >
    <h3
      className="flex-all"
    >
      Export complaints
    </h3>
    <button
      className="a-btn a-btn__link"
      data-gtm_ignore="true"
    >
      Close
      <svg
        className="cf-icon-svg"
        viewBox="0 0 1000 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
        />
      </svg>
    </button>
  </div>
  <div
    className="body"
  >
    <div
      className="body-copy instructions"
    >
      To download a copy of this dataset, choose the file format and which complaints you want to export below.
    </div>
    <div
      className="group"
    >
      <div
        className="group-title"
      >
        Select a format for the exported file
        <span
          className="required-helper"
        >
           (required)
        </span>
      </div>
      <div
        className="body-copy"
      >
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="format_csv"
            onChange={[Function]}
            type="radio"
            value="csv"
          />
          <label
            className="a-label"
            htmlFor="format_csv"
          >
            CSV
          </label>
        </div>
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="format_json"
            onChange={[Function]}
            type="radio"
            value="json"
          />
          <label
            className="a-label"
            htmlFor="format_json"
          >
            JSON
          </label>
        </div>
      </div>
    </div>
    <div
      className="heres-the-url"
    >
      <h4>
        Link to your complaint search results for future reference
      </h4>
      <div
        className="layout-row"
      >
        <input
          className="flex-all a-text-input"
          id="exportUri"
          readOnly={true}
          type="text"
        />
        <button
          className="a-btn a-btn__secondary"
          disabled={true}
          onClick={[Function]}
        >
          <div>
            <span
              className="a-btn_icon"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 686.6 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M526.4 162.5H144.7c-16.3 0-29.7 13.4-29.7 29.7v77.9H29.7C13.4 270.1 0 283.5 0 299.8v719.4c0 16.3 13.4 29.7 29.7 29.7h512.2c16.3 0 29.7-13.4 29.7-29.7v-77.9h85.3c16.3 0 29.7-13.4 29.7-29.7V322.7L526.4 162.5zm-351.4 60h308v110.8c0 3.6.6 7 1.7 10.2l-73.4-73.4H175v-47.6zm336.6 766.3H60V330.1h308v110.8c0 17.5 14.3 31.8 31.8 31.8h111.8v516.1zm115-107.6h-55V430.3l-66.9-66.9c3.2 1.1 6.6 1.7 10.2 1.7h111.7v516.1z"
                />
              </svg>
            </span>
            Copy
          </div>
        </button>
      </div>
    </div>
    <div
      className="timeliness-warning"
    >
      The export process could take several minutes if you're downloading many complaints
    </div>
  </div>
  <div
    className="footer layout-row"
  >
    <button
      className="a-btn"
      data-gtm_ignore="true"
      disabled={true}
      onClick={[Function]}
    >
      Start export
    </button>
    <button
      className="a-btn a-btn__link a-btn__warning"
      data-gtm_ignore="true"
    >
      Cancel
    </button>
  </div>
</section>
`;

exports[`component::DataExport initial state renders without crashing 1`] = `
<section
  className="export-modal"
>
  <div
    className="header layout-row"
  >
    <h3
      className="flex-all"
    >
      Export complaints
    </h3>
    <button
      className="a-btn a-btn__link"
      data-gtm_ignore="true"
    >
      Close
      <svg
        className="cf-icon-svg"
        viewBox="0 0 1000 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
        />
      </svg>
    </button>
  </div>
  <div
    className="body"
  >
    <div
      className="body-copy instructions"
    >
      To download a copy of this dataset, choose the file format and which complaints you want to export below.
    </div>
    <div
      className="group"
    >
      <div
        className="group-title"
      >
        Select a format for the exported file
        <span
          className="required-helper"
        >
           (required)
        </span>
      </div>
      <div
        className="body-copy"
      >
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="format_csv"
            onChange={[Function]}
            type="radio"
            value="csv"
          />
          <label
            className="a-label"
            htmlFor="format_csv"
          >
            CSV
          </label>
        </div>
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="format_json"
            onChange={[Function]}
            type="radio"
            value="json"
          />
          <label
            className="a-label"
            htmlFor="format_json"
          >
            JSON
          </label>
        </div>
      </div>
    </div>
    <div
      className="group"
    >
      <div
        className="group-title"
      >
        Select which complaints you'd like to export
        <span
          className="required-helper"
        >
           (required)
        </span>
      </div>
      <div
        className="body-copy"
      >
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="dataset_filtered"
            onChange={[Function]}
            type="radio"
            value="filtered"
          />
          <label
            className="a-label"
            htmlFor="dataset_filtered"
          >
            <div
              className="multiline-label"
            >
              <div>
                Filtered dataset (
                <span>
                  1,001
                </span>
                 complaints)
              </div>
              <div
                className="body-copy"
              >
                (only the results of the last search and/or filter)
              </div>
            </div>
          </label>
        </div>
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="dataset_full"
            onChange={[Function]}
            type="radio"
            value="full"
          />
          <label
            className="a-label"
            htmlFor="dataset_full"
          >
            <div
              className="multiline-label"
            >
              <div>
                Full dataset (
                <span>
                  9,999
                </span>
                 complaints)
              </div>
              <div
                className="body-copy"
              >
                (not recommended due to very large file size)
              </div>
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      className="heres-the-url"
    >
      <h4>
        Link to your complaint search results for future reference
      </h4>
      <div
        className="layout-row"
      >
        <input
          className="flex-all a-text-input"
          id="exportUri"
          readOnly={true}
          type="text"
        />
        <button
          className="a-btn a-btn__secondary"
          disabled={true}
          onClick={[Function]}
        >
          <div>
            <span
              className="a-btn_icon"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 686.6 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M526.4 162.5H144.7c-16.3 0-29.7 13.4-29.7 29.7v77.9H29.7C13.4 270.1 0 283.5 0 299.8v719.4c0 16.3 13.4 29.7 29.7 29.7h512.2c16.3 0 29.7-13.4 29.7-29.7v-77.9h85.3c16.3 0 29.7-13.4 29.7-29.7V322.7L526.4 162.5zm-351.4 60h308v110.8c0 3.6.6 7 1.7 10.2l-73.4-73.4H175v-47.6zm336.6 766.3H60V330.1h308v110.8c0 17.5 14.3 31.8 31.8 31.8h111.8v516.1zm115-107.6h-55V430.3l-66.9-66.9c3.2 1.1 6.6 1.7 10.2 1.7h111.7v516.1z"
                />
              </svg>
            </span>
            Copy
          </div>
        </button>
      </div>
    </div>
    <div
      className="timeliness-warning"
    >
      The export process could take several minutes if you're downloading many complaints
    </div>
  </div>
  <div
    className="footer layout-row"
  >
    <button
      className="a-btn"
      data-gtm_ignore="true"
      disabled={true}
      onClick={[Function]}
    >
      Start export
    </button>
    <button
      className="a-btn a-btn__link a-btn__warning"
      data-gtm_ignore="true"
    >
      Cancel
    </button>
  </div>
</section>
`;

exports[`component::DataExport visual changes switches the button after a copy 1`] = `
<section
  className="export-modal"
>
  <div
    className="header layout-row"
  >
    <h3
      className="flex-all"
    >
      Export complaints
    </h3>
    <button
      className="a-btn a-btn__link"
      data-gtm_ignore="true"
    >
      Close
      <svg
        className="cf-icon-svg"
        viewBox="0 0 1000 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
        />
      </svg>
    </button>
  </div>
  <div
    className="body"
  >
    <div
      className="body-copy instructions"
    >
      To download a copy of this dataset, choose the file format and which complaints you want to export below.
    </div>
    <div
      className="group"
    >
      <div
        className="group-title"
      >
        Select a format for the exported file
        <span
          className="required-helper"
        >
           (required)
        </span>
      </div>
      <div
        className="body-copy"
      >
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={true}
            className="a-radio"
            id="format_csv"
            onChange={[Function]}
            type="radio"
            value="csv"
          />
          <label
            className="a-label"
            htmlFor="format_csv"
          >
            CSV
          </label>
        </div>
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="format_json"
            onChange={[Function]}
            type="radio"
            value="json"
          />
          <label
            className="a-label"
            htmlFor="format_json"
          >
            JSON
          </label>
        </div>
      </div>
    </div>
    <div
      className="group"
    >
      <div
        className="group-title"
      >
        Select which complaints you'd like to export
        <span
          className="required-helper"
        >
           (required)
        </span>
      </div>
      <div
        className="body-copy"
      >
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={false}
            className="a-radio"
            id="dataset_filtered"
            onChange={[Function]}
            type="radio"
            value="filtered"
          />
          <label
            className="a-label"
            htmlFor="dataset_filtered"
          >
            <div
              className="multiline-label"
            >
              <div>
                Filtered dataset (
                <span>
                  1,001
                </span>
                 complaints)
              </div>
              <div
                className="body-copy"
              >
                (only the results of the last search and/or filter)
              </div>
            </div>
          </label>
        </div>
        <div
          className="m-form-field m-form-field__radio m-form-field__lg-target"
        >
          <input
            checked={true}
            className="a-radio"
            id="dataset_full"
            onChange={[Function]}
            type="radio"
            value="full"
          />
          <label
            className="a-label"
            htmlFor="dataset_full"
          >
            <div
              className="multiline-label"
            >
              <div>
                Full dataset (
                <span>
                  9,999
                </span>
                 complaints)
              </div>
              <div
                className="body-copy"
              >
                (not recommended due to very large file size)
              </div>
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      className="heres-the-url"
    >
      <h4>
        Link to your complaint search results for future reference
      </h4>
      <div
        className="layout-row"
      >
        <input
          className="flex-all a-text-input"
          id="exportUri"
          readOnly={true}
          type="text"
        />
        <button
          className="a-btn export-url-copied"
          disabled={true}
          onClick={[Function]}
        >
          <div>
            <span
              className="a-btn_icon"
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 1000 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm259 284.2L481.4 870.3c-8.2 14.1-22.7 23.4-39 24.8-1.4.1-2.9.2-4.3.2-14.8 0-28.9-6.6-38.4-18L244.4 690.9c-17.9-21-15.4-52.6 5.7-70.5 21-17.9 52.6-15.4 70.5 5.7.2.3.5.5.7.8l109.4 131.4 241.8-418.8c13.6-24 44.2-32.4 68.2-18.8 24 13.6 32.4 44.2 18.8 68.2l-.5.5z"
                />
              </svg>
            </span>
            Copied
          </div>
        </button>
      </div>
    </div>
    <div
      className="timeliness-warning"
    >
      The export process could take several minutes if you're downloading many complaints
    </div>
  </div>
  <div
    className="footer layout-row"
  >
    <button
      className="a-btn"
      data-gtm_ignore="true"
      disabled={true}
      onClick={[Function]}
    >
      Start export
    </button>
    <button
      className="a-btn a-btn__link a-btn__warning"
      data-gtm_ignore="true"
    >
      Cancel
    </button>
  </div>
</section>
`;
